chromium_src/chrome/installer/setup/install_worker_unittest.cc

```diff
@@ -0,0 +1,180 @@
+      : root_(root) {
+    EXPECT_TRUE(nt::SetTestingOverride(root_, path));
+  }
+  ~ScopedNTRegistryTestingOverride() {
+    nt::SetTestingOverride(root_, std::wstring());
+  }
+
+ private:
+  nt::ROOT_KEY root_;
+};
+
+// based on chrome/installer/setup/install_worker_unittest.cc
+class MockWorkItemList : public WorkItemList {
+ public:
+  MockWorkItemList() {}
+
+  MOCK_METHOD5(AddCopyTreeWorkItem,
+               WorkItem*(const base::FilePath&,
+                         const base::FilePath&,
+                         const base::FilePath&,
+                         CopyOverWriteOption,
+                         const base::FilePath&));
+  MOCK_METHOD1(AddCreateDirWorkItem, WorkItem*(const base::FilePath&));
+  MOCK_METHOD3(AddCreateRegKeyWorkItem,
+               WorkItem*(HKEY, const std::wstring&, REGSAM));
+  MOCK_METHOD3(AddDeleteRegKeyWorkItem,
+               WorkItem*(HKEY, const std::wstring&, REGSAM));
+  MOCK_METHOD4(
+      AddDeleteRegValueWorkItem,
+      WorkItem*(HKEY, const std::wstring&, REGSAM, const std::wstring&));
+  MOCK_METHOD2(AddDeleteTreeWorkItem,
+               WorkItem*(const base::FilePath&, const base::FilePath&));
+  MOCK_METHOD4(AddMoveTreeWorkItem,
+               WorkItem*(const base::FilePath&,
+                         const base::FilePath&,
+                         const base::FilePath&,
+                         MoveTreeOption));
+  // Workaround for gmock problems with disambiguating between string pointers
+  // and DWORD.
+  WorkItem* AddSetRegValueWorkItem(HKEY a1,
+                                   const std::wstring& a2,
+                                   REGSAM a3,
+                                   const std::wstring& a4,
+                                   const std::wstring& a5,
+                                   bool a6) override {
+    return AddSetRegStringValueWorkItem(a1, a2, a3, a4, a5, a6);
+  }
+
+  MOCK_METHOD6(AddSetRegStringValueWorkItem,
+               WorkItem*(HKEY,
+                         const std::wstring&,
+                         REGSAM,
+                         const std::wstring&,
+                         const std::wstring&,
+                         bool));
+  MOCK_METHOD6(AddSetRegDwordValueWorkItem,
+               WorkItem*(HKEY,
+                         const std::wstring&,
+                         REGSAM,
+                         const std::wstring&,
+                         DWORD,
+                         bool));
+};
+
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
+
+// Test for when registry key exists and value is already `1`.
+TEST_F(BraveSetupInstallWorkerTest, CleanupAlreadyRan) {
+  registry_util::RegistryOverrideManager override_manager;
+  std::wstring temp;
+  ASSERT_NO_FATAL_FAILURE(
+      override_manager.OverrideRegistry(HKEY_LOCAL_MACHINE, &temp));
+  ScopedNTRegistryTestingOverride nt_override(nt::HKLM, temp);
+
+  // Write out a value `1` (simulating already ran)
+  base::win::RegKey key(HKEY_LOCAL_MACHINE,
```

Comment:
I would extract the logic for writing `kBraveVpnOneTimeServiceCleanupStoragePath` into a function, because it is used multiple times in this file.

Reply:
I'm not sure that makes sense - it's helpful to have the handle... (so value can be re-checked after the test) and there's not really a good way to handle that. Each read or write is basically one line- the creation of the handle to the key is also one line.